### PR TITLE
Conform multiphase CFL to expectations

### PR DESCRIPTION
**Pull request by mbkuhn**
## Summary

Multiplies advective CFL by 2 near interface so that different constraints, in different parts of the domain, can apply together.

## Pull request type

Please check the type of change introduced:

- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Checklist

The following is included:

- [ ] new unit-test(s)
- [ ] new regression test(s)
- [ ] documentation for new capability

This PR was tested by running:

- the unit tests
  - [ ] on GPU 
  - [x] on CPU 
- the regression tests
  - [ ] on GPU 
  - [x] on CPU 

## Additional background

The volume-of-fluid scheme implemented in AMR-Wind requires a CFL of < 0.5 to ensure conservation of each phase. Therefore, it has been the recommendation to run multiphase simulations at CFL < 0.5. However, for simulations where there are dynamics both near and far from the interface, such as an ABL simulation with waves, this may be unnecessarily restrictive, e.g., when the flow in the air is at a higher CFL compared to the flow near the waves. The changes in this PR multiply the advective CFL calculated near the interface by 2, so it can be compared with 1 instead of with 0.5. This will prevent dt from being overly restricted by the CFL in a multiphase ABL case.

### CODE REVIEW

**Comment by mbkuhn** [comment]
The reg tests have been updated to reflect this new usage of CFL for multiphase cases. These changes will only affect the reg test results in multiphase cases with a variable dt. From among those run by ctest, this is only the inertial_drop case, and I have confirmed that this is the only one that changes.

File: amr-wind/incflo_compute_dt.cpp [hunk 1]
```diff
@@ -112,6 +112,11 @@ void incflo::ComputeDt(bool explicit_diffusion)
                         result = std::abs(v_bx(i, j, k, 0)) * dxinv[0] / fac_x +
                                  std::abs(v_bx(i, j, k, 1)) * dxinv[1] / fac_y +
                                  std::abs(v_bx(i, j, k, 2)) * dxinv[2] / fac_z;
+
+                        // Multiply advective CFL by 2 when near interface
+                        result *= 2.0;
```
Comment (gdeskos) [inline amr-wind/incflo_compute_dt.cpp:117]: Hmm is this 0.5, to do with the mass-momentum consistency criterion?

Comment (mbkuhn) [reply amr-wind/incflo_compute_dt.cpp:117]: pretty much -- the VOF scheme needs CFL < 0.5 to ensure conservation of mass. By multiplying this (the calculated advective CFL) by 2, that criterion becomes CFL < 1.0.

Essentially, with this change, the user can enforce CFL < 0.5 near the interface and enforce CFL < 1 elsewhere by setting the max CFL to 1 in the input file.

Comment (gdeskos) [reply amr-wind/incflo_compute_dt.cpp:117]: okay, make sure you remove any restrictions for CFL=0.5 from other parts of the code.

Comment (mbkuhn) [reply amr-wind/incflo_compute_dt.cpp:117]: That's what I did with the reg tests. This change doesn't apply to the CFL calculation when the velocities are prescribed, so I didn't change the reg tests with prescribed velocities.

**Review by gdeskos — APPROVED** [review]

**Pull request merged**